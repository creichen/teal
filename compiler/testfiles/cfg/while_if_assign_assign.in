fun f(a) = {
  while a {
    if a {
      a := 2;
    }
    a := 0;
  }
}

