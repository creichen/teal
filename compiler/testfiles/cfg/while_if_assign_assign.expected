Entry#0 @ 1 :
  -> Access#0
VarDecl#0 @ 1 :
Access#0 @ 2 :
  -> Access#1
  -> Exit#0
Access#1 @ 3 :
  -> IntConstant#0
  -> IntConstant#1
Access#2 @ 4 :
  -> AssignStmt#0
AssignStmt#0 @ 4 :
  -> IntConstant#1
IntConstant#0 @ 4 :
  -> Access#2
Access#3 @ 6 :
  -> AssignStmt#1
AssignStmt#1 @ 6 :
  -> Access#0
IntConstant#1 @ 6 :
  -> Access#3
Exit#0 @ 8 :
